apps/challenges/aws_utils.py

```diff
@@ -1004,6 +1004,133 @@ def create_eks_nodegroup(challenge, cluster_name):
     create_service_by_challenge_pk(client, challenge_obj, client_token)
 
 
+def delete_multiple_challenge_evaluation_clusters(queryset):
+    """
+    The function called by the admin action method to delete all the challenge evaluation clusters used by the
+    selected challenges. This does not delete the roles, subnets, route tables, security groups, etc... associated with
+    these clusters.
+
+    Parameters:
+    queryset (<class 'django.db.models.query.QuerySet'>): The queryset of selected challenges in the django admin page.
+
+    Returns:
+    dict: keys-> 'count': the number of workers successfully stopped.
+                 'failures': a dict of all the failures with their error messages and the challenge pk
+    """
+    from .models import ChallengeEvaluationCluster
+
+    if settings.DEBUG:
+        failures = []
+        for challenge in queryset:
+            failures.append(
+                {
+                    "message": "Challenge evaluation clusters cannot be deleted in development environment",
+                    "challenge_pk": challenge.pk,
+                }
+            )
+        return {"count": 0, "failures": failures}
+
+    count = 0
+    failures = []
+    for challenge in queryset:
+        try:
+            ChallengeEvaluationCluster.objects.get(challenge=challenge)
+        except ChallengeEvaluationCluster.DoesNotExist:
+            response = "Please select challenges with active evaluation clusters only."
+            failures.append(
+                {"message": response, "challenge_pk": challenge.pk}
+            )
+            continue
```

Comment:
Can you instead check if `challenge.workers` count is null for this check. The `ChallengeEvaluationCluster` object will exist for each code upload challenge so this check will never fail

Reply:
Okay. Should we not delete a challenge's ChallengeEvaluationCluster when its cluster is successfully deleted in that case?

Reply:
I noticed that we do check for a ChallengeEvaluationCluster's existence at times. https://github.com/Cloud-CV/EvalAI/blob/master/apps/challenges/views.py#L2835
Furthermore, evaluation clusters are only created for challenges that are docker based. Let's keep the response in there

Reply:
Done